src/vaadin-grid.html

```diff
@@ -770,9 +777,19 @@
       _resizeHandler() {
         this._updateDetailsCellHeights();
         this._accessIronListAPI(super._resizeHandler, true);
+        this._updateScrollerMeasurements();
         this._updateHeaderFooterMetrics();
       }
 
+      _updateScrollerMeasurements() {
```

Comment:
Shouldn't this be at `vaadin-grid-scroll-mixin` as it is also using this method?

Reply:
It's called within `_resizeHandler` right before this method. I've decided that it would be easier to declare it over here, but can be changed.

Reply:
If the cached object is only accessed in `vaadin-grid-scroll-mixin` it would be cleaner to place it there to have the code properly organized.

Have you verified that caching this actually optimizes performance? if not, then it's premature and shouldn't be done. There are numerous occasions when scroll width of the grid might change and since we'd need to have this cache always in sync, `_updateScrollerMeasurements` should be invoked in all those places.

Reply:
hmm.. good point. If they are only being used at `_translateStationaryElements`, it might make sense to use the `outerscroller` values straight from there.

Reply:
I have moved `_updateScrollerMeasurements` to the `vaadin-grid-scroll-mixin` - https://github.com/vaadin/vaadin-grid/pull/1706/commits/1253f93e54455be1f178d83284c5bc530637b709

As for the optimization for the caching, I have verified it with the number of calls to calculate value out from `outerScroller.scrollWidth` and `outerScroller.clientWidth` (which is currently done in `_updateScrollerMeasurements`). 

If we are going to do it in `_translateStationaryElements` the amount of those calls will be approximately doubled as it is called from `_scrollHandler` -> `_afterScroll`. 

For the cases I have checked the `_resizeHandler` (where `_updateScrollerMeasurements` is called) is covering the occasions when scroll width of the grid is changing and we need to sync.

The optimization of this behavior is coming from the caution note mentioned in the comment above (https://github.com/vaadin/vaadin-grid/pull/1706#discussion_r400191427). It's not removing all of the calls, but still reduces those amount. Can get rid out of it in order to avoid any possible issues with changes of scroll width as anyway the performance is affected by extra calculations compared to `ltr` mode.

Reply:
My earlier statement: "...might cause some performance impact" meant that in theory there might be some performance hit with obtaining scrollWidth on every scroll event. But it was still hypothetical, not a fact, so a verification would have been preferred before optimizing. I think it's ok to go with this suggested approach as the optimization is quite trivial.